lib/reviewlette.rb

```diff
@@ -52,23 +51,47 @@ def check_repo(repo_name, token)
       end
 
       puts "Found matching trello card: #{card.name}"
-      reviewers = select_reviewers(card, how_many_should_review(card))
+
+      assignees = issue[:assignees].map { |a| a[:login] }
+      already_assigned_members = @members.select { |m| assignees.include? m['github_username'] }
+      wanted_number = how_many_should_review(issue_flags)
+
+      reviewers = if assignees.size > wanted_number
+                    change_in_reviewers = true
+                    already_assigned_members[0...wanted_number]
+                  elsif assignees.size < wanted_number
+                    change_in_reviewers = true
+                    select_reviewers(card, wanted_number, already_assigned_members)
+                  else
+                    change_in_reviewers = false
+                    already_assigned_members
+                  end
 
       if reviewers.empty?
         puts "Could not find a reviewer for card: #{card.name}"
         next
       end
 
-      repo.add_assignee(issue_id, reviewers.first['github_username'])
-      repo.comment_reviewers(issue_id, reviewers, card)
-
-      @trello.comment_reviewers(card, repo_name, issue_id, reviewers)
+      if change_in_reviewers
+        repo.add_assignees(issue_id, reviewers.map { |r| r['github_username'] } )
+        repo.comment_reviewers(issue_id, reviewers, card)
+        @trello.comment_reviewers(card, repo_name, issue_id, reviewers)
+      end
       @trello.move_card_to_list(card, 'In review')
     end
   end
 
-  def select_reviewers(card, number = 1)
-    reviewers = @members['members']
+  def scan_flags(issue_title)
+    flags = issue_title[/\A\[(.*)\]/, 1]
+    if flags
+      flags.split(',').map(&:strip)
```

Comment:
Are you sure that the flags should be comma-separated?
`[~!*] Foo` looks better for me than `[~, !, *] Foo`, and maybe even `[DONTMERGE, !] Foo`